app/Http/Controllers/api/RegisterController.php

```diff
@@ -7,57 +8,34 @@
 use App\Http\Requests\RegisterRequest;
 use Illuminate\Http\JsonResponse;
 use Illuminate\Support\Facades\DB;
-use Illuminate\Support\Facades\Log;
 use App\Service\User\UserRegisterService;
 use Exception;
-use PDOException;
 
 class RegisterController extends Controller
 {
 
-    private UserRegisterService $userService;
+    private UserRegisterService $userRegisterService;
 
-    public function __construct(UserRegisterService $userService)
+    public function __construct(UserRegisterService $userRegisterService)
     {
-        $this->userService = $userService;
+        $this->userRegisterService = $userRegisterService;
     }
 
     public function register(RegisterRequest $request): JsonResponse
     {
-        $dataRegister = $request->only(['username', 'dni', 'email', 'specialization', 'password']);
+        $input = $request->only(['username', 'dni', 'email', 'specialization', 'password']);
 
         DB::beginTransaction();
 
         try {
-            $result = $this->userService->createUser($dataRegister);
-
-            if ($result !== false) {
-                DB::commit();
-                return response()->json([
-                    'message'=> 'User registered successfully.',
-                    'email'=> $result['email'],
-                    'token'=> $result['token']
-                ], 200);
-            } else {
-                DB::rollBack();
-                return response()->json([
-                    'message'=> 'ProcessFailed. User register Failed', 400
-                ]);
-            }
-        } catch (PDOException $e) {
-            DB::rollBack();
-            Log::error('Database error during user registration:', [
-                'exception' => $e->getMessage(),
-                'trace' => $e->getTraceAsString(),
-            ]);
-            return response()->json('Database error occurred during registration.', 500);
+            $result = $this->userRegisterService->createUser($input);
+
+            DB::commit();
+
+            return response()->json($result, 200);
         } catch (Exception $e) {
```

Comment:
Specify the catch for \DomainException cases

Reply:
Cuando es \DomainException, se supone que estamos gestionando el error nosotros. Por tanto, viene con un "$e->getCode()" que hemos preparado para que coincida con un error de tipo HTTP. Por tanto, se puede hacer esto:

```php
catch (\DomainException $e) {
            DB::rollBack();
            Log::error('Domain exception:', [
                'exception' => $e->getMessage(),
                'trace' => $e->getTraceAsString(),
            ]);
            return response()->json(null, $e->getCode()); // We want the error is only shown in log report
        }
```

Reply:
done